padl/graph.py

```diff
@@ -0,0 +1,664 @@
+"""Node and Graph classes
+
+Graph and Node are to be used inside Transform and not independently.
+"""
+
+from typing import Callable, Iterable, Iterator, List, Optional, Set, Tuple, Union
+from IPython.display import Image
```

Comment:
I don't think we have `IPython` as one of our dependencies. Is it possible to remove this dependency?